extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -165,6 +173,19 @@ public void handle(HttpServerRequest httpServerRequest) {
                 httpServerRequest.response().setStatusCode(503).end();
             }
         }
+
+        private boolean uriValid(HttpServerRequest httpServerRequest) {
+            if (DISABLE_URI_VALIDATION) {
+                return true;
+            }
+            try {
+                // we simply need to know if the URI is valid
+                new URI(httpServerRequest.uri());
+                return true;
```

Comment:
Technically a more accurate validation would be: 

```suggestion
            return uri.getScheme() != null && uri.getHost() != null;
```

Reply:
Very true, but that's not what we do know, so best leave it as is 😎

Reply:
That would prevent entries like `127.0.0.1` (without the `http://`) to be considered valid